athena-gcs/src/main/java/com/amazonaws/athena/connectors/gcs/GcsMetadataHandler.java

```
@@ -0,0 +1,320 @@
+     * corresponding the Athena catalog that was queried.
+     */
+    @Override
+    public ListSchemasResponse doListSchemaNames(BlockAllocator allocator, ListSchemasRequest request) throws Exception
+    {
+        Set<String> schema = new LinkedHashSet<>();
+        if (glueClient != null) {
+            try {
+                schema.addAll(super.doListSchemaNames(allocator, request, DB_FILTER).getSchemas());
+            }
+            catch (RuntimeException e) {
+                LOGGER.warn("doListSchemaNames: Unable to retrieve schemas from AWSGlue.", e);
+            }
+        }
+        return new ListSchemasResponse(request.getCatalogName(), schema);
+    }
+
+    /**
+     * Used to get the list of tables that this source contains.
+     *
+     * @param allocator Tool for creating and managing Apache Arrow Blocks.
+     * @param request   Provides details on who made the request and which Athena catalog and database they are querying.
+     * @return A ListTablesResponse which primarily contains a List<TableName> enumerating the tables in this
+     * catalog, database tuple. It also contains the catalog name corresponding the Athena catalog that was queried.
+     */
+    @Override
+    public ListTablesResponse doListTables(BlockAllocator allocator, final ListTablesRequest request) throws Exception
+    {
+        Set<TableName> tables = new LinkedHashSet<>();
+        if (glueClient != null) {
+            try {
+                tables.addAll(super.doListTables(allocator,
+                        new ListTablesRequest(request.getIdentity(), request.getQueryId(), request.getCatalogName(),
+                                request.getSchemaName(), null, UNLIMITED_PAGE_SIZE_VALUE),
+                        TABLE_FILTER).getTables());
+            }
+            catch (RuntimeException e) {
+                LOGGER.warn("doListTables: Unable to retrieve tables from AWSGlue in database/schema {}", request.getSchemaName(), e);
+            }
```

Same comment here about not eating the exception.

In general, please address all the situations where you are catching and eating exceptions.

---

all try catches are removed.